### PR TITLE
Query: Re-enable tests for Issue#16088

### LINKED ISSUE #16088: Query: Nav expansion failed to expand some navigations
```C#
[ConditionalTheory]
        [MemberData(nameof(IsAsyncData))]
        public virtual Task Manually_created_left_join_propagates_nullability_to_navigations(bool isAsync)
        {
            return AssertQuery<Level1, Level2>(
                isAsync,
                (l1s, l2s) =>
                    from l1_manual in l1s
                    join l2_manual in l2s on l1_manual.Id equals l2_manual.Level1_Optional_Id into grouping
                    from l2_manual in grouping.DefaultIfEmpty()
                    where l2_manual.OneToOne_Required_FK_Inverse2.Name != "L3 02"
                    select l2_manual.OneToOne_Required_FK_Inverse2.Name,
                (l1s, l2s) =>
                    from l1_manual in l1s
                    join l2_manual in l2s on l1_manual.Id equals l2_manual.Level1_Optional_Id into grouping
                    from l2_manual in grouping.DefaultIfEmpty()
                    where Maybe(l2_manual, () => l2_manual.OneToOne_Required_FK_Inverse2.Name) != "L3 02"
                    select Maybe(l2_manual, () => l2_manual.OneToOne_Required_FK_Inverse2.Name));
        }
```
`OneToOne_Required_FK_Inverse2` is not expanded into join.

### PR DESCRIPTION
**Pull request by smitpatel**
Fixed in new nav expansion

Close #16088

### CODE REVIEW

File: test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs [hunk 1]
```diff
@@ -5124,7 +5124,7 @@ select new
                     from t in ts
                     join g in gs on t.GearNickName equals g.Nickname into grouping
                     from g in grouping.DefaultIfEmpty()
-                    where !MaybeScalar<bool>(g, () => g.HasSoulPatch) == true || g == null
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs]: @maumar - Verify this test please. The client side query seemed to be different from server side hence I changed.

Comment (maumar) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs]: this looks like a behavior change from what we do previously, look at the sql that we used to generate

Comment (smitpatel) [reply test/EFCore.Specification.Tests/Query/GearsOfWarQueryTestBase.cs]: I will investigate

**Comment by smitpatel** [comment]
Ping @maumar

**Pull request merged**